edx_exams/apps/api/v1/views.py

```diff
@@ -808,3 +827,37 @@ def get(self, request, course_id):
         """
         allowances = StudentAllowance.get_allowances_for_course(course_id)
         return Response(AllowanceSerializer(allowances, many=True).data)
+
+    def post(self, request, course_id):  # pylint: disable=unused-argument
+        """
+        HTTP POST handler. Creates allowances based on the given list.
+        """
+        allowances = request.data
+
+        serializer = AllowanceSerializer(data=allowances, many=True)
+
+        if serializer.is_valid():
+            allowance_objects = [
+                StudentAllowance(
+                    user=(
+                        User.objects.get(username=allowance['username'])
+                        if allowance.get('username')
+                        else User.objects.get(email=allowance['email'])
+                    ),
+                    exam=Exam.objects.get(id=allowance['exam_id']),
```

Comment:
Normally I'd say don't query inside a loop but it's probably okay here. Probably a waste of complexity optimizing it. It could be worth at least commenting that we expect the number of allowances to be reasonably small.